subgraphs/uniswap-v3/src/common/creators.ts

```diff
@@ -0,0 +1,325 @@
+  pool.name = protocol.name + " " + LPtoken.symbol;
+  pool.symbol = LPtoken.symbol;
+
+  poolAmounts.inputTokens = [token0.id, token1.id];
+  poolAmounts.inputTokenBalances = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+
+  // Used to track the number of deposits in a liquidity pool
+  let poolDeposits = new _HelperStore(poolAddress);
+  poolDeposits.valueInt = INT_ZERO;
+
+  // create the tracked contract based on the template
+  PoolTemplate.create(Address.fromString(poolAddress));
+
+  pool.save();
+  token0.save();
+  token1.save();
+  LPtoken.save();
+  poolAmounts.save();
+  poolDeposits.save();
+}
+
+function createPoolFees(poolAddressString: string, fee: i64): string[] {
+  // LP Fee
+  let poolLpFee = new LiquidityPoolFee("lp-fee-" + poolAddressString);
+  poolLpFee.feeType = LiquidityPoolFeeType.FIXED_LP_FEE;
+  poolLpFee.feePercentage = convertFeeToPercent(fee);
+
+  // Protocol Fee
+  let poolProtocolFee = new LiquidityPoolFee("protocol-fee-" + poolAddressString);
+  poolProtocolFee.feeType = LiquidityPoolFeeType.FIXED_PROTOCOL_FEE;
+  poolProtocolFee.feePercentage = PROTOCOL_FEE_TO_OFF;
+
+  // Trading Fee
+  let poolTradingFee = new LiquidityPoolFee("trading-fee-" + poolAddressString);
+  poolTradingFee.feeType = LiquidityPoolFeeType.FIXED_TRADING_FEE;
+  poolTradingFee.feePercentage = convertFeeToPercent(fee);
+
+  poolLpFee.save();
+  poolProtocolFee.save();
+  poolTradingFee.save();
+
+  return [poolLpFee.id, poolProtocolFee.id, poolTradingFee.id];
+}
+
+// Update store that tracks the deposit count per pool
+function incrementDepositHelper(poolAddress: string): void {
+  let poolDeposits = _HelperStore.load(poolAddress)!;
+  poolDeposits.valueInt = poolDeposits.valueInt + INT_ONE;
+  poolDeposits.save();
+}
+
+export function createDeposit(event: ethereum.Event, amount0: BigInt, amount1: BigInt): void {
+  let poolAddress = event.address.toHexString();
+
+  let pool = getLiquidityPool(poolAddress);
+  let poolAmounts = getLiquidityPoolAmounts(poolAddress);
+
+  let protocol = getOrCreateDex();
+
+  let token0 = getOrCreateToken(pool.inputTokens[0]);
+  let token1 = getOrCreateToken(pool.inputTokens[1]);
+
+  // Convert tokens according to decimals
+  let amount0Converted = convertTokenToDecimal(amount0, token0.decimals);
+  let amount1Converted = convertTokenToDecimal(amount1, token1.decimals);
+
+  // Get the value in USD of the deposit
+  let amountUSD = amount0Converted.times(token0.lastPriceUSD!).plus(amount1Converted.times(token1.lastPriceUSD!));
+
+  // reset tvl aggregates until new amounts calculated
+  protocol.totalValueLockedUSD = protocol.totalValueLockedUSD.minus(pool.totalValueLockedUSD);
+
+  // Update pool balances adjusted for decimals and not adjusted
+  pool.inputTokenBalances = [pool.inputTokenBalances[0].plus(amount0), pool.inputTokenBalances[1].plus(amount1)];
+  poolAmounts.inputTokenBalances = [poolAmounts.inputTokenBalances[0].plus(amount0Converted), poolAmounts.inputTokenBalances[1].plus(amount1Converted)];
```

Comment:
Just a stylistic thing, usually in Typescript/JS this is written as:
```
poolAmounts.inputTokenBalances = [
  poolAmounts.inputTokenBalances[0].plus(amount0Converted), 
  poolAmounts.inputTokenBalances[1].plus(amount1Converted)
];
```
for better readability

Reply:
Updated